node/src/effect.rs

```diff
@@ -254,11 +254,20 @@ pub trait EffectOptionExt {
     ///
     /// The function `f_some` is used to translate the returned value from an effect into an event,
     /// while the function `f_none` does the same for a returned `None`.
-    fn option<U, F, G>(self, f_some: F, f_none: G) -> Effects<U>
+    fn fold<U, F, G>(self, f_some: F, f_none: G) -> Effects<U>
```

Comment:
Shouldn't it rather be `map_or_else`, like [the corresponding method on `Option`](https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or_else)?

Reply:
Also, we should probably switch the order of its arguments, so it's in line with the standard library.

Reply:
I think this ordering of arguments is better and the standard library should change it :)

Reply:
Done in [`4e187e2` (#635)](https://github.com/CasperLabs/casper-node/pull/635/commits/4e187e27d16153b0f933af336f0abae1cdb7263c)